### PR TITLE
touchpoints app configuration

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue

Adds Touchpoints to the app and resolves [3626](https://github.com/CDCgov/prime-simplereport/issues/3626)

Available on pentest here: https://pentest.simplereport.gov/

## Changes Proposed

- Adds Touchpoints to the app
- Moves the tab button on mobile to the drawer menu to not block PXP

## Testing

- View the deployed test instance and verify everything works
 
## Screenshots / Demos

- For large changes, please pair with a designer to ensure changes are as intended

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### CODE REVIEW

**Comment by fzhao99** [comment]
@kenieh if you can double check the implementation of the mobile button is what you expected, that'd be great!

File: frontend/package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "start-js": "react-scripts start",
     "prebuild": "yarn compile-scss",
     "build": "INLINE_RUNTIME_CHUNK=false REACT_APP_CURRENT_COMMIT=$(git rev-parse HEAD) react-scripts build",
-    "postbuild": "echo 'pushstate: enabled' > build/Staticfile && echo $(git rev-parse HEAD) > build/static/commit.txt",
+    "postbuild": "echo 'pushstate: enabled' > build/Staticfile && echo $(git rev-parse HEAD) > build/static/commit.txt && cd build && mv *touchpoints*.js static/",
```
Comment (fzhao99) [inline frontend/package.json:60]: adding in this piece to get around some weirdness with the way our Azure setup is interacting with Create React App.

**Review by emyl3 — COMMENTED** [review]
Looks good to me! Thank you for your work on this! 😸 

Only strange thing I noticed was the checkmark for the checkbox is gone when I select it. Is this intentional?
Screen Shot 2022-06-14 at 3 52 11 PM

**Comment by johanna-skylight** [comment]
It looks good!.  I noticed a few styling weirdness but I am not sure if that's how the style works for the static site:
1- Radio buttons that are checked look different. They don't have the white space that surrounds the blue circle.
Screen Shot 2022-06-15 at 9 19 20 AM

2- Checked checkbox does not display the white check.
Screen Shot 2022-06-15 at 9 19 11 AM


3- For the mobile view the tab at the bottom should be centered and take most of the with (similar to the form). Right now it looks fixed to the left. I think a media query for mobile with below properties will fix it.

```
width: calc(100% - 24px);
left: 12px;
```
**Example of centered tab**
Screen Shot 2022-06-15 at 9 27 45 AM

**Comment by fzhao99** [comment]
> Looks good to me! Thank you for your work on this! 😸
> 
> Only strange thing I noticed was the checkmark for the checkbox is gone when I select it. Is this intentional? Screen Shot 2022-06-14 at 3 52 11 PM

Thanks for calling this out Elisa! I went ahead and fixed it.

**Comment by fzhao99** [comment]
> It looks good!. I noticed a few styling weirdness but I am not sure if that's how the style works for the static site: 1- Radio buttons that are checked look different. They don't have the white space that surrounds the blue circle. Screen Shot 2022-06-15 at 9 19 20 AM
> 
> 2- Checked checkbox does not display the white check. Screen Shot 2022-06-15 at 9 19 11 AM
> 
> 3- For the mobile view the tab at the bottom should be centered and take most of the with (similar to the form). Right now it looks fixed to the left. I think a media query for mobile with below properties will fix it.
> 
> ```
> width: calc(100% - 24px);
> left: 12px;
> ```
> 
> **Example of centered tab** Screen Shot 2022-06-15 at 9 27 45 AM

so the static site touchpoints implementation happens in an entirely different repo 😅

Those styles are here: https://github.com/CDCgov/prime-simplereport-site/blob/main/_sass/touchpoints.scss

I'll made a ticket [here](https://github.com/CDCgov/prime-simplereport/issues/3895) with these issues and we can prioritize those as necessary.

File: frontend/src/app/analytics/TouchpointsButton.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import "./TouchpointsButton.scss";
+
+const TouchpointsButton: React.FC = () => {
+  return (
+    <div className="touchpoints-wrapper sr-app-touchpoints-button">
+      {" "}
```
[the remark this answers is not in the document]
Comment (fzhao99) [reply frontend/src/app/analytics/TouchpointsButton.tsx]: So I think this was an addition by the linter, but I went ahead and deleted it.

**Review by emyl3 — APPROVED** [review]
Nice work on this! Thank you for opening the issue to address the styling! 😸

**Review by johanna-skylight — APPROVED** [review]

**Pull request merged**